homeassistant/components/elmax/switch.py

```diff
@@ -13,34 +16,52 @@
 from .common import ElmaxEntity
 from .const import DOMAIN
 
+_LOGGER = logging.getLogger(__name__)
+
 
 class ElmaxSwitch(ElmaxEntity, SwitchEntity):
     """Implement the Elmax switch entity."""
 
     @property
     def is_on(self) -> bool:
         """Return True if entity is on."""
-        if self.transitory_state is not None:
-            return self.transitory_state
-        return self._device.opened
+        return self.coordinator.get_actuator_state(self._device.endpoint_id).opened
+
+    async def _wait_for_state_change(self, timeout=5.0) -> bool:
+        """Refresh data and wait until the state state changes."""
+        start_ts = time.time()
+        old_state = self.coordinator.get_actuator_state(self._device.endpoint_id).opened
+        new_state = old_state
+        while new_state == old_state:
+            # Check for timeout
+            if (time.time() - start_ts) > timeout:
+                _LOGGER.warning(
+                    "Timeout occurred while waiting for state change from Elmax cloud"
+                )
+                return False
+            # Otherwise sleep a bit and then trigger an update
+            await asyncio.sleep(0.5)
```

Comment:
Fetching new data every 0.5 seconds for five seconds is very frequent. We normally don't allow polling more frequent than every 5 seconds.

Can we refresh once after some time and then just wait for another 5 seconds if we're not ready and refresh again a final time? Ie make it max two refreshes.

Reply:
Hi @MartinHjelmare , 
at the beginning I wanted to do as you suggested (~2s). After having a look at the Verisure component, I realized that component was [polling for "changes" on a specific transaction ID every 0.5 seconds](https://github.com/home-assistant/core/blob/dev/homeassistant/components/verisure/alarm_control_panel.py#L66). That is why I decided to do the same. I know: refreshing the entire state of the coordinator might be more expensive than polling on a specific transaction ID, but on the other hand we have no "transaction-id" on Elmax to poll. 
As you can see, that polling value was not invented by myself, but inspired to Verisure component.

Do you think we shall change both components or can we be consistent and leave 0.5s polling interval?

Reply:
I don't think the verisure code is relevant for this discussion. Just adjust here.

Reply:
Hi @MartinHjelmare ,
I've just done exactly what you asked for.